src/compiler/checker.ts

```diff
@@ -60,6 +72,35 @@ namespace ts {
         undefinedSymbol.declarations = [];
         const argumentsSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.Transient, "arguments");
 
+        const {typeToString, getSymbolDisplayBuilder, symbolToString, signatureToString, typePredicateToString, visibilityToString} =
+            createWriter(
+                createSymbol,
+                getSymbolOfNode,
+                compilerOptions,
+                languageVersion,
+                getTypeParametersOfClassOrInterface,
+                getAccessibleSymbolChain,
+                needsQualification,
+                getQualifiedLeftMeaning,
+                hasExternalModuleSymbol,
+                isTypeAny,
+                () => nothingType,

```

Comment:
done. It means I have to repeat the names while defining `createWriter`, but that's not too bad:

```ts
createWriter({f, g, h}: { f: () => number, ... }) ...
```